subprojects/base-services/src/main/java/org/gradle/internal/reflect/ReflectionCache.java

```diff
@@ -34,12 +34,14 @@ public T get(Class<?> key) {
         synchronized (lock) {
             cached = cache.get(key);
         }
-        if (cached != null) {
+        if (cached != null && !hasExpired(cached)) {
```

Comment:
Ok this just happened, I need to find a better way... Hard without creating hard references between everything :(